qiskit/pulse/passmanager.py

```diff
@@ -0,0 +1,226 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Passmanager module for pulse schedules."""
+
+import contextlib
+import logging
+import time
+from typing import Union, List, Set
+
+from qiskit import pulse
+from qiskit.pulse.basepasses import BasePass
+from qiskit.pulse import exceptions
+from qiskit.pulse.states import State
+
+logger = logging.getLogger(__name__)
+
+
+class PassManager:
```

Comment:
I'm sure you wouldn't do this if it wasn't necessary. Can you explain why the existing one won't work here, too?

Reply:
The transpiler passmanager is A. in the `transpiler` module and B. baked into its core its run method accepts `List[QuantumCircuit, QuantumCircuit]`, it also assumes calls to `runningpassmanager` which only operates on `DAGCircuit`. Either we will need to maintain separate compiler infrastructure, or make the top-level inputs to the infrastructure something more generic then a `QuantumCircuit` such as a `State` that is generated by the overarching `Compiler` which accepts a `Program` or `Module`. I think its kind of possible to see the start of this in this prototype.